--- a/src/ccsd/moints_trp.F
+++ b/src/ccsd/moints_trp.F
@@ -651,6 +651,14 @@
       call ga_sync
       my_id = ga_nodeid()
       call ga_distribution(g_exch, my_id, rlo, rhi, clo, chi )
+      if (rhi .lt. rlo) then
+         rlo = 1
+         rhi = nbf * nvir
+      endif
+      if (chi .lt. clo) then
+         clo = 1
+         chi = 1
+      endif
 #ifdef BAD_GACCESS
       ld=rhi-rlo+1
       if(.not.ma_push_get(MT_DBL,ld,
