--- /home/rgoswami/Git/Github/OmniPotentRPC/rgpot/third_party/nwchem/src/ccsd/moints_trp.F.orig_classic	2026-06-30 17:35:03.431655943 -0500
+++ /home/rgoswami/Git/Github/OmniPotentRPC/rgpot/third_party/nwchem/src/ccsd/moints_trp.F	2026-06-30 17:35:52.702734666 -0500
@@ -651,6 +651,14 @@
       call ga_sync
       my_id = ga_nodeid()
       call ga_distribution(g_exch, my_id, rlo, rhi, clo, chi )
+      if (rhi .lt. rlo) then
+         rlo = 1
+         rhi = nbf * nvir
+      endif
+      if (chi .lt. clo) then
+         clo = 1
+         chi = 1
+      endif
 #ifdef BAD_GACCESS
       ld=rhi-rlo+1
       if(.not.ma_push_get(MT_DBL,ld,
